pkg/sdk/applications_def.go

```diff
@@ -0,0 +1,142 @@
+package sdk
+
+import g "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk/poc/generator"
+
+//go:generate go run ./poc/main.go
+
+/*
+ * 	todo: add definition for `CREATE APPLICATION <name> FROM LISTING <listing_name> [ COMMENT = '<string_literal>' ] [ WITH TAG ( <tag_name> = '<tag_value>' [ , ... ] ) ]`
```

Comment:
Why we are leaving it for later? Can we add it to the SNOW-933289 epic as a subtask if we don't need it now?